### ISSUE TITLE
Modal dialogs can be bypassed with keyboard navigation

### ISSUE THREAD

**Issue by pfattisc** [MahApps/MahApps.Metro#2400]
## What steps will reproduce this issue?

While a modal dialog is shown (i.e. _MetroWindow.ShowMetroDialogAsync()_ )
it is possible to switch keyboard focus to the metro window by pressing _**Ctrl + Tab**_ or _**Ctrl + Shift + Tab**_
Furthermore all CommandBindings and InputBindings of the mainwindow are still working (e.g. _Ctrl + S_ for the Save command).
### Expected outcome

Keyboard navigation to the mainwindow should be impossible while a modal dialog is shown.
Maybe an attached behaviour (MetroWindow) should make it possible to remove/restore the CommandBindings and InputBindings while a dialog is shown.
## 
### Environment
- MahApps.Metro **v1.2.4**
- Windows **7**
- Visual Studio **2013**
- .NET Framework **4.5**

**Comment by punker76** [comment]
@pfattisc KeyboardNavigations are set to Cycle

```xaml
        <Setter Property="KeyboardNavigation.ControlTabNavigation" Value="Cycle" />
        <Setter Property="KeyboardNavigation.DirectionalNavigation" Value="Cycle" />
        <Setter Property="KeyboardNavigation.TabNavigation" Value="Cycle" />
```

CommandBindings and InputBindings still an issue

**Comment by pfattisc** [comment]
Should work for me.
I solved the CommandBindings & InputBindings issue by adding/removing them whenever the _DialogManager DialogOpened/DialogClosed Event_ occurs.

**Comment by frankschierle** [comment]
There is a problem with access keys (mnemonics), too. Pressing an access key that is assigned to a control in the MetroWindow will switch keyboard focus to the MetroWindow, while a dialog is shown.

Furthermore the access keys of the dialog buttons are not working if the same access keys are used for a control inside the MetroWindow.

I created a small repository to reproduce the error (press ALT+N while the dialog is shown):

https://github.com/frankschierle/MahAppsAccessKeyBug

__Expected outcome__
The access key should trigger the dialog button while the dialog is shown.
